Update operational workflow to work with GSI develop branch

ADCollard opened the pull request:
Update operational workflow to work with GSI develop branch. This mostly involves changing the links to elements that have been moved out of the GSI repository since GFSv16.3.  The ultimate aim is to be able to cycle with v16 workflow but using a version of GSI close to develop. This will not reproduce operations.

Resolves #3045 
  
# Type of change
- [ ] Bug fix (fixes something broken)
- [x] New feature (adds functionality)
- [x] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? YES
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? YES 
  - [x] GSI <!-- NOAA-EMC/GSI#1234 -->

# How has this been tested?
Testing is limited to WCOSS2.   Initialised full resolution cycling experiment.  Will compare to operational performance.

# Checklist
- [ ] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] This change is covered by an existing CI test or a new one has been added
- [ ] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [ ] I have made corresponding changes to the system documentation if necessary

ADCollard committed:
Add GSI script files into workflow scripts directory

ADCollard committed:
Check out GSI develop

ADCollard committed:
Check out GSI develop

ADCollard committed:
Add Enkf scripts and jobs

ADCollard committed:
Turn off some links

ADCollard committed:
Add former gsi.fd scripts to ush

ADCollard committed:
Add external gsi_utils

ADCollard committed:
Update some links

ADCollard committed:
Update gsi_utils version number and add ncdiag

ADCollard committed:
Use an earlier version of gsi_utils

ADCollard committed:
Remove use_qsatensmean ref. GSI#602

ADCollard committed:
Slight change to version of GSI_utils

ADCollard committed:
Another version number change

ADCollard committed:
Fix some links

ADCollard committed:
Merge branch 'dev/gfs.v16.GSIdev' of https://github.com/ADCollard/global-workflow into dev/gfs.v16.GSIdev

ADCollard committed:
Another link fix

ADCollard committed:
Merge tag 'gfs.v16.3.19' into dev/gfs.v16.GSIdev

ADCollard committed:
Fix small bug in archive and change default ANAVINFO

ADCollard committed:
Turn on optconv option - see GSI #375

ADCollard committed:
Restore observation monitoring capability

ADCollard committed:
More radmon updates

ADCollard committed:
Add monitor scripts from global-workflow #f159d39

ADCollard committed:
Fix info messages

ADCollard committed:
Slightly earlier tag for GSI-Monitor

ADCollard committed:
Merge branch 'dev/gfs.v16.GSIdev' of https://github.com/ADCollard/global-workflow into dev/gfs.v16.GSIdev

ADCollard committed:
Fix conflicts

KateFriedman-NOAA commented on jobs/JGDAS_ATMOS_VERFOZN:
```diff
@@ -0,0 +1,92 @@
+# Set up environment for GDAS Ozone Monitor job
+#############################################################
+source "${HOMEgfs}/ush/preamble.sh"
+source "${HOMEgfs}/ush/jjob_header.sh" -e "vrfy" -c "base vrfy"
```
This looks like the script in `develop` for GFSv17 since it has the preamble and jjob_header scripts in here. Those scripts won't exist in v16 so we need the version of this script as it is in v16 (or somewhere between).

KateFriedman-NOAA commented on jobs/JGDAS_ATMOS_VERFRAD:
```diff
@@ -0,0 +1,107 @@
+# Set up environment for GDAS Radiance Monitor job
+#############################################################
+source "${HOMEgfs}/ush/preamble.sh"
+source "${HOMEgfs}/ush/jjob_header.sh" -e "vrfy" -c "base vrfy"
```
Same comment as above, need to use the v16 versions with any needed changes for GSI `develop`.

KateFriedman-NOAA commented on jobs/JGDAS_ATMOS_VMINMON:
```diff
@@ -0,0 +1,79 @@
+# GDAS Minimization Monitor (MinMon) job
+###########################################################
+source "${HOMEgfs}/ush/preamble.sh"
+source "${HOMEgfs}/ush/jjob_header.sh" -e "vrfy" -c "base vrfy"
```
Same comment as above, need to use the v16 versions with any needed changes for GSI `develop`.

emilyhcliu commented:
@ADCollard I see that most file changes are due to the changes in the .gitignore.  
These changes make sense.  I see that the CRTM is still crtm 2.4.1.
I am checking out the branch to build.
Two questions. 
- Will this global-workflow branch build GSI from the GSI develop? 
- Do you want me to run tests with this global-workflow branch?

ADCollard commented:
> @ADCollard I see that most file changes are due to the changes in the .gitignore. These changes make sense. I see that the CRTM is still crtm 2.4.1. I am checking out the branch to build. Two questions.
> 
> * Will this global-workflow branch build GSI from the GSI develop?
> * Do you want me to run tests with this global-workflow branch?

@emilyhcliu Yes, the idea is that this is a v16 workflow that will checkout the develop version of GSI (eventually it will be a specific tag).  

I am mostly looking for feedback on whether this is the right way to do this.   I have also started a real time parallel on WCOSS2 (this will only run on WCOSS2), but right now the machine is unavailable, so it might be easiest just to look at those results.

Yes, the CRTM version is 2.4.0.1 which is the current operational version.   I don't think we have time to move to CRTM v3.0 for GFSv16.4, but that would be addressed via a different issue anyway.

Thanks for looking at this!

KateFriedman-NOAA commented on jobs/JGFS_ATMOS_VMINMON:
```diff
@@ -0,0 +1,82 @@
+###########################################################
+# GFS Minimization Monitor (MinMon) job
+###########################################################
+source "${HOMEgfs}/ush/preamble.sh"
+source "${HOMEgfs}/ush/jjob_header.sh" -e "vrfy" -c "base vrfy"
```
Same comment as above, need to use the v16 versions with any needed changes for GSI `develop`.

KateFriedman-NOAA commented on scripts/exgdas_atmos_verfozn.sh:
```diff
@@ -0,0 +1,91 @@
+#! /usr/bin/env bash
+
+source "$HOMEgfs/ush/preamble.sh"
```
Same comment as above. Can't use preamble but thankfully it looks like this script hasn't changed much between v16 and v17 so you can likely keep most of the v17 version:
```
[kfriedma@hercules-login-2 scripts]$ pwd
/work/noaa/global/kfriedma/git_hercules/dev-gfs.v16.GSIdev/scripts
[kfriedma@hercules-login-2 scripts]$ diff ../../release-gfs.v16.3.21/scripts/exgdas_atmos_verfozn.sh exgdas_atmos_verfozn.sh
1c1
< #/bin/sh
---
> #! /usr/bin/env bash
3c3
< set -ax
---
> source "$HOMEgfs/ush/preamble.sh"
12,13d11
< export scr=exgdas_vrfyozn.sh
<
73c71
<    count=`ls diag* | grep ".nc4" | wc -l`
---
>    count=$(ls diag* | grep ".nc4" | wc -l)
76,77c74,75
<       for filenc4 in `ls diag*nc4.gz`; do
<          file=`echo $filenc4 | cut -d'.' -f1-2`.gz
---
>       for filenc4 in $(ls diag*nc4.gz); do
>          file=$(echo $filenc4 | cut -d'.' -f1-2).gz
92,98d89
<
< if [[ "$VERBOSE" = "YES" ]]; then
<    echo "end exgdas_vrfyozn.sh, exit value = ${err}"
< fi
<
<
< set +x
```

KateFriedman-NOAA commented on scripts/exgdas_atmos_verfrad.sh:
```diff
@@ -0,0 +1,222 @@
+#! /usr/bin/env bash
+
+source "$HOMEgfs/ush/preamble.sh"
```
Same comment. Can likely ingest the v17 copy here but need to remove the preamble.

KateFriedman-NOAA commented on scripts/exgdas_atmos_vminmon.sh:
```diff
@@ -0,0 +1,119 @@
+#! /usr/bin/env bash
+
+source "$HOMEgfs/ush/preamble.sh"
```
Same comment, remove the preamble line but can most likely keep the rest of the v17 updates in this script.

KateFriedman-NOAA commented on scripts/exgfs_atmos_vminmon.sh:
```diff
@@ -0,0 +1,116 @@
+#! /usr/bin/env bash
+
+source "$HOMEgfs/ush/preamble.sh"
```
Same comment as above, remove preamble, can keep rest of the v17 updates.

KateFriedman-NOAA commented on sorc/checkout.sh:
```diff
@@ -32,14 +32,41 @@ fi
-    git clone --recursive --branch gfsda.v16.3.20 https://github.com/NOAA-EMC/GSI.git gsi.fd >> ${topdir}/checkout-gsi.log 2>&1
+#    git clone --recursive --branch gfsda.v16.3.12 https://github.com/NOAA-EMC/GSI.git gsi.fd >> ${topdir}/checkout-gsi.log 2>&1
+# Check out develop for now
+    git clone --recursive https://github.com/NOAA-EMC/GSI.git gsi.fd >> ${topdir}/checkout-gsi.log 2>&1
```
Leave note to make sure this gets updated to a tag eventually.

KateFriedman-NOAA commented on sorc/checkout.sh:
```diff
@@ -32,14 +32,41 @@ fi
+# Check out a version before the changes for Thompson microphysics were introduced.
+    git clone https://github.com/NOAA-EMC/GSI-Utils.git gsi_utils.fd >> ${topdir}/checkout-gsi_utils.log 2>&1
+    cd gsi_utils.fd
+    git checkout 2a15d3b514cb05a9c1343e437f134375ad260369
```
Leaving note to make sure this gets updated to a tag eventually.

KateFriedman-NOAA commented on sorc/checkout.sh:
```diff
@@ -32,14 +32,41 @@ fi
+# Check out a version before changes for the new directory structure were introduced.
+    git clone https://github.com/NOAA-EMC/GSI-Monitor.git gsi_monitor.fd >> ${topdir}/checkout-gsi_monitor.log 2>&1
+    cd gsi_monitor.fd
+    git checkout e1f9f21af16ce912fdc2cd75c5b27094a550a0c5
```
Leaving note to make sure this gets updated to a tag eventually.

KateFriedman-NOAA commented on sorc/link_fv3gfs.sh:
```diff
@@ -91,41 +91,33 @@ cd ${pwd}/../util               ||exit 8
         $LINK ../sorc/ufs_utils.fd/util/$file .
     done
 
+#-----------------------------------
+#--add gfs_wafs link if checked out
+if [ -d ${pwd}/gfs_wafs.fd ]; then
+#-----------------------------------
+ cd ${pwd}/../jobs               ||exit 8
+     $LINK ../sorc/gfs_wafs.fd/jobs/*                         .
+ cd ${pwd}/../parm               ||exit 8
+     [[ -d wafs ]] && rm -rf wafs
+    $LINK ../sorc/gfs_wafs.fd/parm/wafs                      wafs
+ cd ${pwd}/../scripts            ||exit 8
+    $LINK ../sorc/gfs_wafs.fd/scripts/*                      .
+ cd ${pwd}/../ush                ||exit 8
+    $LINK ../sorc/gfs_wafs.fd/ush/*                          .
+ cd ${pwd}/../fix                ||exit 8
+    [[ -d wafs ]] && rm -rf wafs
+    $LINK ../sorc/gfs_wafs.fd/fix/*                          .
+fi
```
This WAFS block shouldn't be getting added. Need to recheck the sync with the `release/gfs.v16.3.21` branch and/or remove this block in your branch. The `release/gfs.v16.4` branch will need to get synced with the prior release branches again anyway but let's remove this block now.

KateFriedman-NOAA commented on sorc/link_fv3gfs.sh:
```diff
@@ -91,41 +91,33 @@ cd ${pwd}/../util               ||exit 8
-    $LINK ../sorc/gsi.fd/ush/calcanl_gfs.py      .
-    $LINK ../sorc/gsi.fd/ush/calcinc_gfs.py      .
-    $LINK ../sorc/gsi.fd/ush/getncdimlen         .
+# The following are now explicitly in the global-workflow repository
+#    $LINK ../sorc/gsi.fd/ush/gsi_utils.py        .
+#    $LINK ../sorc/gsi.fd/ush/calcanl_gfs.py      .
+#    $LINK ../sorc/gsi.fd/ush/calcinc_gfs.py      .
+#    $LINK ../sorc/gsi.fd/ush/getncdimlen         .
```
Good to know, let's remove these lines now.

KateFriedman-NOAA commented on ush/jjob_header.sh:
```diff
@@ -0,0 +1,115 @@
+#! /usr/bin/env bash
```
Please remove this script from your branch. This won't go into the GFS with v16 and still needs sign off from NCO for v17.

KateFriedman-NOAA commented on ush/preamble.sh:
```diff
@@ -0,0 +1,92 @@
+#! /usr/bin/env bash
```
Please remove this from your branch. This won't go into the GFS with v16 and still needs sign off from NCO for v17.

KateFriedman-NOAA commented on ush/rstprod.sh:
```diff
@@ -0,0 +1,19 @@
+#! /usr/bin/env bash
+
+source "$HOMEgfs/ush/preamble.sh"
```
If we definitely need this script from the v17 package then remove the preamble line.

KateFriedman-NOAA reviewed (changes requested):
We can't bring in most v17 script updates and need to stick closer to the v16 versions for this update. Please see my comments and let me know if I can assist at all. Thanks @ADCollard !

ADCollard commented:
> We can't bring in most v17 script updates and need to stick closer to the v16 versions for this update. Please see my comments and let me know if I can assist at all. Thanks @ADCollard !

@KateFriedman-NOAA Thank you for this review.  This is exactly the type of feedback I was looking for.

Just so I am clear on this, why are the `preamble.sh` and `jjob_header.sh` scripts forbidden in v16?  I will attempt to make these scripts look closer to v17, but I want to be sure if there is something particularly problematic with these.

KateFriedman-NOAA commented:
> Just so I am clear on this, why are the preamble.sh and jjob_header.sh scripts forbidden in v16? I will attempt to make these scripts look closer to v17, but I want to be sure if there is something particularly problematic with these.

@ADCollard The preamble and jjob_header scripts are new as of v17 and we have not yet had them signed off on by NCO. They represent a big change to how we structure the JJOB scripts so we need them approved by NCO. Both scripts are also intended to be in every JJOB/scripts script so we shouldn't partially introduce them.

ADCollard commented:
> > Just so I am clear on this, why are the preamble.sh and jjob_header.sh scripts forbidden in v16? I will attempt to make these scripts look closer to v17, but I want to be sure if there is something particularly problematic with these.
> 
> @ADCollard The preamble and jjob_header scripts are new as of v17 and we have not yet had them signed off on by NCO. They represent a big change to how we structure the JJOB scripts so we need them approved by NCO. Both scripts are also intended to be in every JJOB/scripts script so we shouldn't partially introduce them.

@KateFriedman-NOAA OK.  I will remove them.

ADCollard committed:
Update monitor scripts to use versions from GSI #1f6d3aa (6/1/2022)

ADCollard committed:
Update monitor scripts to use versions from GSI #1f6d3aa (6/1/2022)

KateFriedman-NOAA commented on versions/wcoss2.ver:
```diff
@@ -3,7 +3,8 @@ export prod_envir_ver=${prod_envir_ver:-2.0.4} # Allow override from ops ecflow
 
 export obsproc_run_ver=1.3.0
-export prepobs_run_ver=1.2.0
+export prepobs_run_ver=1.1.0
```
Remove the change to `prepobs_run_ver`, it should still be `export prepobs_run_ver=1.2.0`. I'll actually be merging in an update to make it 1.3.0 after GFSv16.3.20 goes into ops in early December. I'll take care of that update when it happens.

ADCollard replied:
Fixed

KateFriedman-NOAA reviewed (changes requested):
Looks good, left one change to make. Can rereview when moved out of draft. Thanks!

ADCollard committed:
Update wcoss2.ver

Correct prepobs  version

ADCollard commented:
> Looks good, left one change to make. Can rereview when moved out of draft. Thanks!

@KateFriedman-NOAA Thanks for looking at this.   Before I take this out of draft I need to test it in a cycled experiment.   I am on foreign travel right now and hence cannot access WCOSS-2.   I will run the test on my return on 11/26/2024.

I am also inviting @EdwardSafford-NOAA to look over what I have done with the monitoring suite.

KateFriedman-NOAA commented:
Sounds good, thanks @ADCollard ! FYI I will be on leave the week of Thanksgiving.

EdwardSafford-NOAA commented:
@ADCollard Monitoring changes look ok, though I think you might need to have `radmon_err_rpt.sh` in the ush directory.  If you point me to your output and/or logs from this branch I can confirm or deny that.

KateFriedman-NOAA committed:
GFSv16.3.20 - annual CO2 update and PlanetIQ assimilation (#3174)

GFSv16.3.20 includes the following updates:

- annual carbon dioxide (CO2) fix file updates for 2025 (within staged fix set on WCOSS2)
- updated GSI global_convinfo.txt to turn on assimilation of PlanetIQ GPS Radio Occultation (GPS-RO) data
- memory increases for a number of jobs (provided by NCO)
- load the cmdaccel module at the start of wave point postprocessor jobs to stabilize runtime (provided by NCO)

Refs #2558

KateFriedman-NOAA committed:
Remake the package for WAFS removal from GFS (#3176)

Remake the release branch for the WAFS removal from the GFS due
to implementation order change. It includes changes previously made in
the original release branch:

- updated release notes
- remove all WAFS scripts and settings
- include resource update for enkfgdas_update job

Refs #2591

YaliMao-NOAA committed:
Updated release note related to WAFS separation changes (#3179)

Refs #2591

KateFriedman-NOAA committed:
Remake the GFS package for obsproc and AFWA upgrade (#3182)

The order of the upgrade to obsproc/v1.3 and the new AFWA global file
was moved back and thus the release branch needed to be remade. This PR
includes all updates for the GFS upgrade to obsproc/v1.3 and the new
global AFWA data file:

- new UFS_UTILS tag (`ops-gfsv16.3.20`) to support new global AFWA file
- new GSI tag (`gfsda.v16.3.22`) to support obsproc/v1.3 and saildrone
- workflow changes to support obsproc/v1.3 (and prepobs/v1.2 internal to
obsproc) and AFWA global file
- also workflow changes to allow developers to run the emcsfc_sfc_prep
job within the rocoto mesh via `DO_SFCPREP=YES` in config.base

This PR is a repeat of prior ones to get this package in place.

Refs #2913

KateFriedman-NOAA committed:
Increase to ncpus for gfs_wave_postpnt job

NCO provided increase in ncpus to avoid ncpu burst warnings.

Refs #2913

KateFriedman-NOAA committed:
Fix for syndat_getjtbul code parsing 3-character storm names (GFSv16.3.21) (#3200)

Update the `sorc/syndat_getjtbul.fd/getjtbul.f` code to resolve a problem parsing 3-character storm names.
This update is being included in the GFSv16.3.21 package (with approval by NCO).

From the developer @Qingfu-Liu:

> The `gfs_atmos_tropcy_qc_reloc` job did not properly parse the vitals
info from a raw bulletin with a tab in the storm name field for storm
"SIX". The code sorc/syndat_getjtbul.fd/getjtbul.f converting a tab to a
space worked well for storms that have at least 4 characters in their
names, but does not work as well for storms with 3-character names. This
code change will fix the parsing problem for 3-character storms.

Refs #2591

KateFriedman-NOAA committed:
Sync merge remote-tracking branch 'origin/release/gfs.v16.3.21' into release/gfs.v16.3.22

* origin/release/gfs.v16.3.21:
  Fix for syndat_getjtbul code parsing 3-character storm names (GFSv16.3.21) (#3200)

KateFriedman-NOAA committed:
GFSv16.3.21 - WAFS removal from GFS (#3281)

The WAFS is now a separate package in operations and is no longer a part of the GFS.

Also included in this package are the following unrelated updates:

1. improvements to the enkfgdas_update job resource and environment settings
2. a fix for the syndat_getjtbul code when parsing 3-character storm names

Refs #2591

AliS-Noaa committed:
Optimized wave point output (#3257)

The `ww3_outp` is optimized to read per time step binary output and create per point spectral and bull/cbull output.

Refs #2913

KateFriedman-NOAA committed:
Merge remote-tracking branch 'origin/dev/gfs.v16' into release/gfs.v16.3.22

* origin/dev/gfs.v16:
  GFSv16.3.21 - WAFS removal from GFS (#3281)

KateFriedman-NOAA committed:
Rename Gulf of Mexico to Gulf of America (#3352)

Update the name of the "Gulf of Mexico" to the "Gulf of America".
This includes fix/parm file updates and a new UWM tag for WW3 rename updates.

Refs #3326

KateFriedman-NOAA committed:
Additional Gulf of Mexico name changes in GFS (#3353)

Make additional "Gulf of Mexico" to "Gulf of America" renames.
The bufr station list is updated and the EMC_verif-global tag
is updated to a version with the same name change.

Refs #3326

KateFriedman-NOAA committed:
Update release notes

Refs #3326

KateFriedman-NOAA committed:
Updated bufr station list for GFS ops (#3350)

Introduces bufr station updates into the GFSv16.3.23 operational
package. There are three updated stations and 121 new stations added to
the bufr station parm files. Changes provided by @BoCui-NOAA .

Refs #2913

KateFriedman-NOAA committed:
Update version to v16.3.23 and sync with new v16.3.22 changes (#3387)

Adjust the package to the new v16.3.23 and incorporate updates
from the new v16.3.22 package (now ahead of this update). 
Additionally, the obsproc (and prepobs) updates are backed 
out for a later upgrade. Now use `obsproc/v1.2.3`.

Refs #2913

KateFriedman-NOAA committed:
Update obsproc_run_ver to v1.2.4

Refs #2913

KateFriedman-NOAA committed:
Add bufr station updates to release notes

Refs #2913

KateFriedman-NOAA committed:
Merge remote-tracking branch 'origin/release/gfs.v16.3.22' into release/gfs.v16.3.23

* origin/release/gfs.v16.3.22:
  Update release notes
  Additional Gulf of Mexico name changes in GFS (#3353)
  Rename Gulf of Mexico to Gulf of America (#3352)

KateFriedman-NOAA committed:
Add missing GSI monitoring ush scripts to link step

- The radmon_diag_ck.sh and rstprod.sh scripts were missing
from the link script step and were causing errors in ops.
- Add the two missing scripts to the link script and
the gitignore file.

Refs #2913

ADCollard committed:
Merged in release/gfs.v16.3.23

ADCollard committed:
Turn on CADS

ADCollard commented:
Code has been updated to [release/gfs.v16.3.23](https://github.com/NOAA-EMC/global-workflow/pull/3068/commits/0fdab56a1709f34d9a2422cc527c8f7995f9a87d). 

The workflow before these changes successfully ran for seven cycles.  I still need to do a longer test with the latest version, but following conversations with @aerorahul, I think this is ready to take out of draft.

Note that CRTMv3 is not longer part of this upgrade.

ADCollard commented:
Actually I have one question.  [Externals.cfg](https://github.com/ADCollard/global-workflow/blob/dev/gfs.v16.GSIdev/Externals.cfg) has the following lines:

```
[GSI]
tag = gfsda.v16.3.22
local_path = sorc/gsi.fd
repo_url = https://github.com/NOAA-EMC/GSI.git
protocol = git
required = True
```

What does this do?   I am checking out the develop version of GSI right now.

KateFriedman-NOAA commented:
> What does this do? I am checking out the develop version of GSI right now.

@ADCollard We aren't using `checkout_externals` anymore but I'm keeping `Externals.cfg` synced with the tags in `sorc/checkout.sh` for consistency. Once you have a GSI tag for `sorc/checkout.sh`, please also set it in `Externals.cfg`. Thanks!

KateFriedman-NOAA commented:
> Code has been updated to [release/gfs.v16.3.23](https://github.com/NOAA-EMC/global-workflow/pull/3068/commits/0fdab56a1709f34d9a2422cc527c8f7995f9a87d).
> 
> The workflow before these changes successfully ran for seven cycles. I still need to do a longer test with the latest version, but following conversations with @aerorahul, I think this is ready to take out of draft.

Awesome, feel free to move this out of draft. Note, I need to make a GFSv16.3.24 package that will have the obsproc/prepobs version updates. I'll let you know when I've done that so you can sync that into the v16.4 release branch or I can do the sync if it happens after this goes into the release branch. Just a heads up.

> Note that CRTMv3 is not longer part of this upgrade.

Noted, thanks!

ADCollard committed:
Merge branch 'release/gfs.v16.4' into dev/gfs.v16.GSIdev

ADCollard commented:
@KateFriedman-NOAA Just to confirm that we will re-tag FV3GFS to 16.4, even though there will not be changes?

KateFriedman-NOAA commented:
> @KateFriedman-NOAA Just to confirm that we will re-tag FV3GFS to 16.4, even though there will not be changes?

The ufs-weather-model was tagged for v16.4 because some WW3 updates were made and requested to be included in v16.4. So the g-w release branch will include the DA updates and a UWM tag update.

ADCollard commented:
@aerorahul Is this enough to kick off phases 1 and 2 that we discussed yesterday.    

As I said then, I am on leave from tonight until 3/14 without WCOSS access, so cannot kick off a parallel before then.   Hopefully we could still do a code handoff around  May 1st.

Thanks.

KateFriedman-NOAA reviewed (approved):
Thanks @ADCollard ! I am going to merge, sync it with v16.3.24 changes, and then review further.
